packages/lib-core/src/utils/__tests__/logger.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { consoleLogger } from '../logger';
+
+describe('consoleLogger', () => {
+  beforeEach(() => {
+    jest.resetModules();
+  });
+
+  test('should have all console loggers', () => {
+    expect(consoleLogger.warn).toBeDefined();
+    expect(consoleLogger.error).toBeDefined();
+    expect(consoleLogger.info).toBeDefined();
+  });
+
+  test('should have all console loggers except info on prod', () => {
+    const consoleSpy = jest.spyOn(console, 'info');
```

Comment:
This can be done via `beforeEach` so `consoleSpy` variable is accessible to all test cases (each `test` or `it` instance).

Reply:
Generally, code that creates a Jest spy/mock should be close to code that resets/teardowns it.

Reply:
Not sure I follow, the spy is watched just in this test so I wouldn't abuse the `beforeEach` for that. However good point of checking the spy in the test before this one.

Reply:
Hmmm... I think I was thinking of a mock impl not a spy 🤔 

I can't find anything that backs up needing to reset a spy when it's in a scoped situation like inside a test.

The scenario I was thinking of was:
```
      // Suppress the console log for a cleaner test
      jest.spyOn(console, 'log').mockImplementation(jest.fn());

     . . . // some test stuff that calls console.log

      // re-enable logs for future tests
      // eslint-disable-next-line no-console
      (console.log as any).mockRestore();
```

Reply:
What I meant is you can move `const consoleSpy = jest.spyOn(console, 'info');` into the outer scope.

It would be created just once, and reset/intitialized before or after each test case.